docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,170 @@
+
+## Summary
+
+This proposal introduces first class support for "unmanaged" infrastructure for CAPI providers to consolidate the boundaries between managed and unmanaged cluster infrastructure.
+
+## Motivation
+
+Currently CAPI providers support an opinionated happy path to create and manage cluster infrastructure lifecycle. The fundamental use case we want to support is bring your own infrastructure. An "unmanaged" CAPI infraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, but CAPI will be able to create compute nodes within it.
+
+This will ease adoption of CAPI in heterogeneous real world environments with restricted privileges and where the provider infrastructure for the cluster needs to be managed out of band.
+
+### Goals
+
+- Introduce support for "unmanaged" infrastructure consistently across CAPI providers.
+- The machine controller must be able to operate and manage machines when the infastructure is "unmanaged".
+
+### Non-Goals/Future Work
+
+- Modify existing managed behaviour.
+
+## Proposal
+
+Providers will add a `ManagementPolicy=Managed|Unmanaged` field to the provider infraCluster spec.
+
+The machine controller must be able to operate without hard dependencies regardless of the cluster infrastructure being managed or unmanaged.
+![](https://i.imgur.com/nA61XJt.png)
+
+### User Stories
+
+#### Story 1 - Alternate control plane provisioning with user managed infrastructure
+As a cluster provider I want to use CAPI in my service offering to orchestrate Kubernetes bootstraping while letting workload cluster operators own their infrastructure lifecycle for the data plane.
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstraping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
+
+#### Story 3 - Consuming existing cloud infrastructure
+As a cluster operator I want to use CAPI to orchestate Kubernetes bootstraping while reusing infrastructure that has already been created in the organisation either by me or another team.
+
+### Implementation Details/Notes/Constraints
+
+**Managed**
+- It will be default and will preserve existing behaviour.
+
+**Unmanaged**
+
+- The provider infraCluster controller will skip any infrastructure reconciliation.
+
+- The provider infraCluster will set readiness to true.
+
+- The provider infraCluster will set a condition unamangedReady to true.
+
+- CAPI will proceed with further reconciliation as usual.
+
+
+### Security Model
+
+When unmanaged no additional privileges for a cloud provider need to be given to CAPI other than the required to manage machines.
+
+#### Multitenancy
+
+CAPI supports a best effort multitenancy model with deploying multiple controllers. In this scenario adhoc privileges for each controller can be granted to satisfy "managed" or "unmanaged" cluster nifrastr
```

Comment:
I'm not sure if this is going to be the case any longer, could you clarify why this requirement is in this proposal? Isn't enough to have the distinction on the InfrastructureCluster level?

Reply:
this is a typo, I don't think we need to mention multitenancy here as it should be transparent https://github.com/kubernetes-sigs/cluster-api/pull/4135#discussion_r569591067